Automattic/woocommerce-payments
# Initial progressive onboarding form

Fixes #5473 

#### Changes proposed in this Pull Request
- Implement `<Stepper/>`, a reusable component to show one step at a time, exposing `nextStep` method and `onComplete` event. It checks for step prop `name` and falls back to `index`, to have a string reference to the step, and use it later for tracking purposes (for example).
- Create steps `Personal`, `Business`, and `Store` details based on the design, with the expected fields and values.
- On Stepper complete event, use `fromDotNotation` on the onboarding data to convert dot notation keys to nested objects, removing undefined/null values. And append it as query to `connectUrl`.
- Use a simple way to render fields which can be extended/refactored to implement validation and tracks, without overengineering it at first.

The following TODO items will be covered as part of other issues:
- #4853 – Populate MCC options and add MCC field test. After we have codes defined on the server.
- #4744 – MCC Select control must provide search functionality and create a notice component.

#### Testing instructions
- Put a breakpoint on the server to debug `$account_data` withing `Onboarding_Controller`, and run the debugger.
- Go to **WCPay Dev Utils** and enable `Force onboarding`, `Enable Progressive Onboarding`, and `Force the plugin to act as disconnected...` if you already have an account connected.
- Navigate to `/wp-admin/admin.php?page=wc-admin&path=/payments/onboarding-prototype`.
- The form should have 3 steps with the fields included in the design.
- Complete every step in the form, checking that business type and company structure and rendered based on the selected country.
- After clicking on `Continue` on the third step, the server debugger should trigger and `$account_data` should contain the values from the filled form fields, similar to:

```
email: "john@doe.com"
business_name: "John Doe LLC"
url: "https://johndoe.com"
country: "US",
company:
    structure: "single_member_llc"
individual:
    first_name: "John"
    last_name: "Doe"
phone: "000000000"
business_type: "company"
annual_revenue: "less_than_250k"
go_live_timeframe: "within_1month"
```

-------------------


**Post merge**
